doc/developers-guide/logging.md

```diff
@@ -0,0 +1,112 @@
+# Logging
+
+To use logger in your module, include
+
+```erlang
+-include("mongoose_logger.hrl").
+```
+
+or
+
+```erlang
+-include("mongoose.hrl").
+```
+
+# Logging macros
+
+There are several macroses:
+
+```erlang
+DEBUG("event=debug_event info=~1000p", [Arg]),
+INFO_MSG("event=info_event info=~1000p", [Arg]),
+ERROR_MSG("event=error_event reason=~1000p", [Arg]),
+WARNING_MSG("event=warning_event reason=~1000p", [Arg]),
+```
+
+Use them to log on corresponding log level.
+
+Don't use `ERROR_MSG` for cases that are not errors.
+
+# Logging levels
+
+A system operator can choose log level by setting loglevel in `ejabberd.cfg`.
+
+- level 5 - debug
+- level 4 - info
+- level 3 - warning
+- level 2 - error
+- level 1 - critical
+- level 0 - none
+
+While there is a macro to log on level 1, we rarely use it.
+
+If an user sets log level 5, than he would see all messages in logs.
+
+Levels 3 or 2 are usually used for production systems.
+
+
+# Logging format
+
+We use modified [logfmt](https://brandur.org/logfmt) format.
+
+This format is [Splunk](https://www.splunk.com/en_us/solutions/solution-areas/log-management.html)
+and [ELK](https://www.elastic.co/elk-stack) friendly.
+
+`event=something_interesting` field is required.
+
+`reason=~1000p` field is commonly used.
```

Comment:
I think we could use an info about default message length limit enforced by lager.